toolchain/check/eval.cpp

```diff
@@ -553,6 +573,62 @@ static auto PerformBuiltinIntComparison(Context& context,
   return MakeBoolResult(context, bool_type_id, result);
 }
 
+// Performs a builtin unary float -> float operation.
+static auto PerformBuiltinUnaryFloatOp(Context& context,
+                                       SemIR::BuiltinFunctionKind builtin_kind,
+                                       SemIR::InstId arg_id)
+    -> SemIR::ConstantId {
+  auto op = context.insts().GetAs<SemIR::FloatLiteral>(arg_id);
+  auto op_val = context.floats().Get(op.float_id);
+
+  switch (builtin_kind) {
+    case SemIR::BuiltinFunctionKind::FloatNegate:
+      op_val = llvm::APFloat(0.0) - op_val;
```

Comment:
```suggestion
      op_val.changeSign();
```
`APFloat(0.0)` will assume IEEE754 64-bit semantics for the `0.0`, which won't be right once we start supporting more kinds of floating-point type.

Reply:
Resolved!